Classes/System/Solr/SolrConnection.php

```diff
@@ -271,136 +281,30 @@ protected function setAuthenticationOnAllEndpoints(Client $client, $username, $p
      */
     protected function getClient(string $endpointKey): Client
     {
-        if ($this->clients[$endpointKey]) {
+        if($this->clients[$endpointKey]) {
             return $this->clients[$endpointKey];
         }
-        // TODO: Should it be a clone of the endpoint? In row 277 the key of the endpoint is set.
-        $endPoint = $this->getNode($endpointKey);
-        print_r($endPoint);
-        $newEndpointOptions = $endPoint->getSolariumClientOptions();
-        $adapter = $this->getClientAdapter($newEndpointOptions, $endpointKey);
-
-        $client = new Client(
-            $adapter,
-            $this->eventDispatcher
-        );
+
+        $adapter = new Psr18Adapter($this->psr7Client, $this->requestFactory, $this->streamFactory);
```

Comment:
This adapter is not compatible with TYPO3 10.4

Guzzle required by Solarium: ^7.2
Guzzle required by TYPO3 10.4: ^6.3.0

There for this is not backportable

Reply:
Guzzle isn't required by solarium:
https://github.com/solariumphp/solarium/blob/f24311b50ead3107c0ceb37cfa41cdace67af431/composer.json#L14-L21
TYPO3 implements PSRs 7/17/18 since TYPO3 10, which can be used by solariums PSR18Adapter.
So the constructor arguments for Solariums Psr18Adapter are (PSR-7-Client, PSR-17s Request factory, PSR-17s StramFactory)

See: https://docs.typo3.org/c/typo3/cms-core/10.2/en-us/Changelog/10.1/Feature-89018-ProvideImplementationForPSR-17HTTPMessageFactories.html?highlight=psr

---

Confusion is maybe the word "Guzzle" on multiple places(Thats because TYPO3 uses Guzzle, and we should use that if possible!).
But see how is psr7Client initialized `$this->psr7Client = $psr7Client ?? GeneralUtility::getContainer()->get(ClientInterface::class);` 
TYPO3s dependency injection does all the things needed for that PSR gibberish.

Reply:
Please change the commit message, since there is the use of the term Guzzle.

This leads to wrong expectation what your commit does.